runtime/oti/j9.h

```diff
@@ -323,11 +323,14 @@ static const struct { \
 #define J9_CLASS_DISALLOWS_LOCKING_FLAGS (J9ClassIsValueType | J9ClassIsValueBased)
 #define J9_CLASS_ALLOWS_LOCKING(clazz) J9_ARE_NO_BITS_SET((clazz)->classFlags, J9_CLASS_DISALLOWS_LOCKING_FLAGS)
 #define J9_IS_J9CLASS_VALUEBASED(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassIsValueBased)
+#define J9_IS_J9CLASS_GENERATED(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassGenerated)
 #ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
 #define J9CLASS_UNPADDED_INSTANCE_SIZE(clazz) J9_VALUETYPE_FLATTENED_SIZE(clazz)
 #define J9_IS_J9CLASS_VALUETYPE(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassIsValueType)
 #define J9_IS_J9CLASS_PRIMITIVE_VALUETYPE(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassIsPrimitiveValueType)
+#define J9_IS_J9CLASS_GENERATED_VT_LTYPE(clazz) (J9_IS_J9CLASS_VALUETYPE(clazz) && J9_IS_J9CLASS_GENERATED(clazz))
 #define J9_IS_J9CLASS_FLATTENED(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassIsFlattened)
+#define J9_ARE_J9CLASSES_SAME(c1, c2) (((c1) == (c2)) || ((c2) == (c1)->Ltype) || ((c2) == (c1)->Qtype))
```

Comment:
This name isn't great - can you find a more descriptive term instead of "same"?

Reply:
That seems reasonable. @tajila what do you think?

Reply:
See discussion below.

Reply:
Updated.

Reply:
Equivalent makes sense